src/tools/rest-backend/authentication_application.cpp

```diff
@@ -0,0 +1,307 @@
+#include <ctime>
+#include <iostream>
+#include <regex>
+#include <stdlib.h>
+#include <string.h>
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <jwt/jwt.h>
+#include <openssl/sha.h>
+
+#include "authentication_application.hpp"
+#include "crypto.hpp"
+#include "root_application.hpp"
+#include "service.hpp"
+#include "user_application.hpp"
+
+namespace kdbrest
+{
+
+/**
+ * @brief the constructor fot the authentication endopint application.
+ * @param srv a service container
+ */
+AuthenticationApp::AuthenticationApp (cppcms::service & srv) : cppcms::application (srv)
+{
+
+	dispatcher ().assign ("", &AuthenticationApp::authenticate, this);
+	mapper ().assign ("");
+}
+
+/**
+ * @brief handler for the authentication resource.
+ *		  attempts an authentication with the submitted credentials. in case the
+ *		  attempt succeeds, a session token will be returned, otherwise an error.
```

Comment:
That's not my indentation, but from the clang reformatter. I think the docu errors are because I changed something, will fix it.